File: Left View of Binary Tree.c
void leftViewUtil(struct Node* root, int level, int* maxLevel) {
    if (root == NULL) return;

    if (*maxLevel < level) {
        printf("%d ", root->data);
        *maxLevel = level;
    }

    leftViewUtil(root->left, level + 1, maxLevel);
    leftViewUtil(root->right, level + 1, maxLevel);
}

void leftView(struct Node* root) {
    int maxLevel = 0;
    leftViewUtil(root, 1, &maxLevel);
}

File: Left View of Binary Tree.py
def leftView(root):
    result = []
    
    def helper(node, level):
        if not node:
            return
        
        if level == len(result):
            result.append(node.data)
        
        helper(node.left, level + 1)
        helper(node.right, level + 1)
    
    helper(root, 0)
    return result
